.github/workflows/pre_release.yaml

```diff
@@ -0,0 +1,53 @@
+name: Canary Release
+
+on:
+  push:
+    branches:
+    - "main"
+    paths-ignore:
+    - "**/*.png"
+
+jobs:
+  canary-release:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Generate Release Manifests
+      run: make generate-manifests IMAGE=envoyproxy/gateway-dev TAG=latest OUTPUT_DIR=release-artifacts
+
+    # Ignore the error from the first time canary release deletion.
+    # We do not have the release at first, after that, the error will not appear again.
+    - name: Delete Canary Release
+      continue-on-error: true
+      run: |
+        gh release delete latest --repo $GITHUB_REPOSITORY
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_REPOSITORY: ${{ github.repository_owner }}/${{ github.event.repository.name }}
+
+    # Same as above, ignore the error from the first time canary tag deletion.
+    # We do not have the tag at first, after that, the error will not appear again.
+    - name: Delete Canary Tag
+      continue-on-error: true
+      run:
+        gh api --method DELETE /repos/$GITHUB_REPOSITORY/git/refs/tags/latest
```

Comment:
thanks for making this change and reusing the `gh` cli in the workflow.

Reply:
`latest` is a special keyword in GH https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases, can we make sure deleting `latest` will delete the `latest` tag and not the latest `vX.Y.Z` tag

Reply:
This is the tag name here:)

Reply:
git/tags/ref/{git-tag-name}